src/marin/run/ray_deps.py

```diff
@@ -0,0 +1,23 @@
+"""Manage Ray runtime dependencies."""
+
+import logging
+
+from ray.runtime_env import RuntimeEnv
+
+logger = logging.getLogger(__name__)
+
+
+def setup_runtime_env_for_packages(
+    runtime_env: RuntimeEnv, extra: list[str] | None = None
+) -> RuntimeEnv:
+    flags = []
+    for ex in extra:
+        if ex.strip():
+            flags.append(f"--extra={ex}")
+
+    # sadly Ray is not capable of accepting it's own attributes to RuntimeEnv so we use a raw dict
+    logger.info(f"Configuring `py_executable` for Ray remote to: {flags}")
+    return dict(
+        py_executable=f"uv run --no-managed-python --frozen --link-mode=copy {' '.join(flags)}",
```

Comment:
previously when i switched us to uv it would be even slower than ray's default and would use more storage. is this solved?